src/Hl7.Fhir.Specification/Specification/Navigation/StructureDefinitionWalker.cs

```diff
@@ -233,20 +233,35 @@ public IEnumerable<StructureDefinitionWalker> Resolve()
         public StructureDefinitionWalker Extension(string url)
         {
             // find the extension children of the current node
-            var extensionChildren = childDefinitions("extension");
+            var extensionChildren = childDefinitions("extension").ToList();
             var selection = extensionChildren.Where(c => isExtensionFor(c, url)).ToList();
 
-            // No children in the current profile -> we can still continue in the definition
-            // of the extension itself.
-            if (selection.Count == 0)
-                return FromCanonical(url);
-            else if (selection.Count == 1)
-                return new StructureDefinitionWalker(selection.Single(), _resolver);
-            else
-                throw new StructureDefinitionWalkerException($"extension('{url}') found multiple extension slices constraining the same extension, with is not allowed for the discriminator at '{Current.CanonicalPath()}'.");
+            return selection switch
+            {
+                // No children in the current profile -> we can still continue in the definition
+                // of the extension itself.
+                { Count: 0 } when isAbsoluteUri(url) => FromCanonical(url),
```

Comment:
wow, nice condition :)

Reply:
Who needs if/else blocks anymore? Let's hope in C#10 they will allow blocks after the '=>' because, that's the only painful reason to use if/else if/else's now.